src/server/.tests/routes/controllers/utilities/passportLoginCllbacks.spec.js

```diff
@@ -0,0 +1,156 @@
+const passportLoginCallbacks = require('../../../../routes/controllers/utilities/passportLoginCallbacks')
+const responseMessages = require('../../../../routes/controllers/utilities/responseMessages')
+const statusCodes = require('../../../../routes/controllers/utilities/statusCodes')
+const EventEmitter = require('events').EventEmitter
+const httpMocks = require('node-mocks-http')
+const assert = require('chai').assert
+let suite = {}
+
+describe('Passport Login Callbacks', () => {
+	beforeEach(() => {
+		suite.responseMock = httpMocks.createResponse({
+			eventEmitter: EventEmitter
+		})
+	})
+
+
+	describe('authenticate callback', () => {
+		beforeEach(() => {
+			suite.authenticateCallback = passportLoginCallbacks.authenticateCallback(null, suite.responseMock)
+		})
+
+		it('should respond with BAD REQUEST error when unknown error was given', done => {
+			
+			//given
+			const DUMMY_ERROR = responseMessages.errors.BAD_REQUEST
+			const DUMMY_USER = null
+			const DUMMY_INFO = null
+
+			//when
+			suite.authenticateCallback(DUMMY_ERROR, DUMMY_USER, DUMMY_INFO)
+
+			//then
+			const responseStatus = suite.responseMock._getStatusCode()
+			const responseBody = suite.responseMock._getData()
+
+			const expectedResponseStatus = statusCodes.BAD_REQUEST
+			const expectedResponseBody = JSON.stringify({
+				description: responseMessages.errors.BAD_REQUEST
+			})
+
+			assert.strictEqual(responseBody, expectedResponseBody)
+			assert.equal(responseStatus, expectedResponseStatus)
+			done()
+		})
+
+		it('should respond with INTERNAL SERVER error when unknown error was given', done => {
```

Comment:
`should respond with BAD REQUEST error when unknown error was given`
`should respond with INTERNAL SERVER error when unknown error was given`
Should it respond with BAD REQUEST, or INTERNAL SERVER ERROR, in case of unknown error?

Reply:
ok